### PR TITLE
Nested hyperlinked urls

### PR DESCRIPTION
**Pull request by ChristianKreuzberger**
I've implemented a url field for nested relations and tests for it.
Referencing Pull Request #79 for this aswell.

### CODE REVIEW

**Comment by ChristianKreuzberger** [comment]
Hi! Just wondering if anyone is looking at this?
@lwm You told me to ping you when I put a PR up, hope this is okay?

File: README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ drf-nested-routers
 =====================
 
 [![Join the chat at https://gitter.im/alanjds/drf-nested-routers](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/alanjds/drf-nested-routers?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
-[![Build Status](https://travis-ci.org/alanjds/drf-nested-routers.svg?branch=master)](https://travis-ci.org/alanjds/drf-nested-routers)
+[![Build Status](https://travis-ci.org/christiankreuzberger/drf-nested-routers.svg?branch=master)](https://travis-ci.org/christiankreuzberger/drf-nested-routers)
```
Comment (lwm) [inline README.md]: You probably want to revert this.

File: rest_framework_nested/relations.py [hunk 2]
```diff
@@ -9,6 +9,40 @@
 import rest_framework.relations
 
 
+class NestedHyperlinkedIdentityField(rest_framework.relations.HyperlinkedIdentityField):
```
Comment (lwm) [inline rest_framework_nested/relations.py]: If this is going into master, we'll probably want to deal with these `FIXME` notes and test them?

File: rest_framework_nested/serializers.py [hunk 3]
```diff
@@ -52,3 +55,37 @@ class Meta:
         field_kwargs = get_nested_relation_kwargs(relation_info)
 
         return field_class, field_kwargs
+
+
+class NestedHyperlinkedIdentityField(HyperlinkedIdentityField):
```
Comment (lwm) [inline rest_framework_nested/serializers.py]: Similar note to above? We'll want to remove those `FIXME` notes and test them.

File: tests/serializers/models.py [hunk 4]
```diff
@@ -41,8 +41,32 @@ class ParentChild2Serializer(nested_serializers.NestedHyperlinkedModelSerializer
...
+
+
+class ParentChild1Serializer2(nested_serializers.NestedHyperlinkedModelSerializer):
+    class Meta:
+        model = Child1
         fields = ('url', 'name')
 
+    url = nested_serializers.NestedHyperlinkedIdentityField(
+        lookup_fields=('parent__pk', 'pk',),
+        view_name='parent3-child1-detail'
```
Comment (lwm) [inline tests/serializers/models.py]: Did you mean `child2` here? Not sure.

File: tests/serializers/test_serializers.py [hunk 5]
```diff
@@ -56,3 +56,8 @@ def test_no_children(self):
         data = self.get_json_response(url)
 
         self.assertEqual(len(data['first']), 0)
+
+    def test_nested_urls(self):
+        url = reverse('parent3-list')
+        data = self.get_json_response(url)
+        print(data)
```
Comment (lwm) [inline tests/serializers/test_serializers.py]: This test doesn't assert anything, so it isn't useful.

File: README.md (hunk 1 above)
Comment (ChristianKreuzberger) [inline README.md]: Agreed

File: tests/serializers/test_serializers.py (hunk 5 above)
Comment (ChristianKreuzberger) [inline tests/serializers/test_serializers.py]: True. It does however fail when you try to serialize the URLs without my Nested Serializer. I will work on improving this test though.

File: tests/serializers/models.py (hunk 4 above)
Comment (ChristianKreuzberger) [inline tests/serializers/models.py]: Pretty sure I meant to use child1 here, and child2 a couple lines below, as I'm using the relations for child1 and child2 in separate viewsets.

File: rest_framework_nested/serializers.py (hunk 3 above)
Comment (ChristianKreuzberger) [inline rest_framework_nested/serializers.py]: The serializer wasn't meant to be in relations anyway. I've removed the code in relations.py, and left it in serializers.py.
The last fixme comment is still here, as I'm not sure which errors could happen, but I will see if I can create some tests for that.

**Comment by ChristianKreuzberger** [comment]
I now feel stupid for not looking at the latest commits... Someone already made this in PR #84, though it does not properly support multiple levels of nested viewsets.

I will see if I can re-use what I already wrote for this.

**Pull request closed without merging**